src/clients/bridges/AdapterManager.ts

```diff
@@ -46,172 +43,37 @@ export class AdapterManager {
-        spokePoolClients,
-        MODE,
-        hubChainId,
-        filterMonitoredAddresses(MODE),
-        logger,
-        SUPPORTED_TOKENS[MODE],
-        constructBridges(MODE),
-        DEFAULT_GAS_MULTIPLIER[MODE] ?? 1
-      );
-    }
-    if (this.spokePoolClients[REDSTONE] !== undefined) {
-      this.adapters[REDSTONE] = new BaseChainAdapter(
-        spokePoolClients,
-        REDSTONE,
-        hubChainId,
-        filterMonitoredAddresses(REDSTONE),
-        logger,
-        SUPPORTED_TOKENS[REDSTONE],
-        constructBridges(REDSTONE),
-        DEFAULT_GAS_MULTIPLIER[REDSTONE] ?? 1
-      );
-    }
-    if (this.spokePoolClients[LISK] !== undefined) {
-      this.adapters[LISK] = new BaseChainAdapter(
-        spokePoolClients,
-        LISK,
-        hubChainId,
-        filterMonitoredAddresses(LISK),
-        logger,
-        SUPPORTED_TOKENS[LISK],
-        constructBridges(LISK),
-        DEFAULT_GAS_MULTIPLIER[LISK] ?? 1
-      );
-    }
-    if (this.spokePoolClients[BLAST] !== undefined) {
-      this.adapters[BLAST] = new BaseChainAdapter(
-        spokePoolClients,
-        BLAST,
-        hubChainId,
-        filterMonitoredAddresses(BLAST),
-        logger,
-        SUPPORTED_TOKENS[BLAST],
-        constructBridges(BLAST),
-        DEFAULT_GAS_MULTIPLIER[BLAST] ?? 1
-      );
-    }
-    if (this.spokePoolClients[SCROLL] !== undefined) {
-      this.adapters[SCROLL] = new ScrollAdapter(logger, spokePoolClients, filterMonitoredAddresses(SCROLL));
-    }
-    if (this.spokePoolClients[CHAIN_IDs.WORLD_CHAIN] !== undefined) {
-      this.adapters[CHAIN_IDs.WORLD_CHAIN] = new BaseChainAdapter(
-        spokePoolClients,
-        CHAIN_IDs.WORLD_CHAIN,
-        hubChainId,
-        filterMonitoredAddresses(CHAIN_IDs.WORLD_CHAIN),
-        logger,
-        SUPPORTED_TOKENS[CHAIN_IDs.WORLD_CHAIN],
-        constructBridges(CHAIN_IDs.WORLD_CHAIN),
-        DEFAULT_GAS_MULTIPLIER[CHAIN_IDs.WORLD_CHAIN] ?? 1
-      );
-    }
-    if (this.spokePoolClients[ZORA] !== undefined) {
-      this.adapters[ZORA] = new BaseChainAdapter(
-        spokePoolClients,
-        ZORA,
-        hubChainId,
-        filterMonitoredAddresses(ZORA),
-        logger,
-        SUPPORTED_TOKENS[ZORA],
-        constructBridges(ZORA),
-        DEFAULT_GAS_MULTIPLIER[ZORA] ?? 1
-      );
-    }
-    if (this.spokePoolClients[ALEPH_ZERO] !== undefined) {
-      this.adapters[ALEPH_ZERO] = new BaseChainAdapter(
-        spokePoolClients,
-        ALEPH_ZERO,
-        hubChainId,
-        filterMonitoredAddresses(ALEPH_ZERO),
-        logger,
-        SUPPORTED_TOKENS[ALEPH_ZERO],
-        constructBridges(ALEPH_ZERO),
-        DEFAULT_GAS_MULTIPLIER[ALEPH_ZERO] ?? 1
-      );
-    }
-
-    if (this.spokePoolClients[INK] !== undefined) {
-      this.adapters[INK] = new BaseChainAdapter(
+    Object.keys(this.spokePoolClients).map((_chainId) => {
+      const chainId = Number(_chainId);
+      assert(chainId.toString() === _chainId);
```

Comment:
If you iterate over the values and dereference the chainId directly this then you can drop the all-keys-are-strings dance:

```suggestion
    Object.values(this.spokePoolClients).map(({ chainId }) => {
```

Reply:
The reason why I didn't do it this way was because our unit tests do not always define a chain id in the spoke pool client, so I assumed that we can't be positive that the chainId field is defined (at least, not in our unit tests).